### PR TITLE
Add Make rules that depend on source material

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

**Comment by stefanv** [comment]
Touching index.Rmd does not trigger a rebuild for some reason.

File: build-requirements.txt [hunk 1]
```diff
@@ -5,3 +5,4 @@
 jupytext
 noteout>=0.0.3a2
 ghp-import
+yaml
```
Comment (matthew-brett) [inline build-requirements.txt:8]: Jupytext already pulls in PyYAML - could we use that instead?

File: source/Makefile [hunk 2]
```diff
@@ -1,40 +1,35 @@
...
 	../scripts/set_version.py $*
 
-html-book-%:
-	$(MAKE) set-version-$*
+../%-book/index.html: set-version-% $(Rmd_sources) $(diagrams)
 	quarto render . --to html
 
-pdf-book-%:
-	$(MAKE) set-version-$*
+../%-book/resampling-with-python.pdf: set-version-% $(Rmd_sources) $(diagrams)
```
Comment (matthew-brett) [inline source/Makefile:23]: Won't the title be `resample_with_r.pdf` for the R edition?

File: source/Makefile [hunk 3]
```diff
@@ -1,40 +1,35 @@
...
+python-html-book: ../python-book/index.html
 
-r-html-book: $(diagrams) html-book-r
+python-pdf-book: ../python-book/resampling-with-python.pdf
 
-r-pdf-book: $(diagrams) pdf-book-r
+r-html-book: ../r-book/index.html
 
-r-all: $(diagrams) all-book-r
+r-pdf-book: ../r-book/resampling-with-python.pdf
```
Comment (matthew-brett) [inline source/Makefile:32]: Ditto previous comment.

File: source/Makefile [hunk 4]
```diff
@@ -1,40 +1,35 @@
...
-
-python-html-book: $(diagrams) html-book-python
-
-python-pdf-book: $(diagrams) pdf-book-python
-
-python-all: $(diagrams) all-book-python
+python-html-book: ../python-book/index.html
 
-r-html-book: $(diagrams) html-book-r
+python-pdf-book: ../python-book/resampling-with-python.pdf
```
Comment (matthew-brett) [inline source/Makefile:28]: Do we need these targets now?  Can't we just use the long form - `../python-book/resampling-with-python.pdf` ?

**Review by matthew-brett — COMMENTED** [review]
A few expressions of confusion.

File: build-requirements.txt (hunk 1 above)
Comment (stefanv) [reply build-requirements.txt:8]: We use it explicitly in our Makefile now, so probably good to make that dependency explicit as well.

File: source/Makefile (hunk 2 above)
Comment (stefanv) [reply source/Makefile:23]: Sloppy, sorry. Fixed.

File: source/Makefile (hunk 4 above)
Comment (stefanv) [reply source/Makefile:28]: It's quite a bit more convenient to type, so perhaps leave for now?

File: build-requirements.txt (hunk 1 above)
Comment (matthew-brett) [reply build-requirements.txt:8]: Any reason not to use PyYAML in the Makefile though?

**Review by matthew-brett — COMMENTED** [review]
Last comment.

I guess there's no way of re-using `%` many times as in `%-book/resampling_with_%.pdf`?

**Comment by stefanv** [comment]
> Last comment.
> 
> I guess there's no way of re-using `%` many times as in `%-book/resampling_with_%.pdf`?

No, unfortunately that does not work.

**Comment by stefanv** [comment]
I keep forgetting that `yaml` is actually `pyyaml`.  Fixed in the requirements file.

**Comment by matthew-brett** [comment]
Thanks for doing that - I find make both boring and intimidating, so it's a relief that you know what you're doing here.

**Comment by stefanv** [comment]
> Thanks for doing that - I find make both boring and intimidating, so it's a relief that you know what you're doing here.

Err... yes... always know exactly what I'm doing when it comes to make.  For fun, you should run `make <target> -d` sometime.

**Pull request merged**